Sources/MongoSwift/Operations/Operation.swift

```diff
@@ -6,3 +6,16 @@ internal protocol Operation {
     /// Executes this operation and returns its corresponding result type.
     func execute() throws -> OperationResult
 }
+
+/// Internal function for generating an options `Document` for passing to libmongoc.
+internal func encodeOptions<T: Encodable>(options: T?,
```

Comment:
since passing the encoder to this function is going to copy it anyway, and we don't care about the strategies when encoding options (and in the future we may introduce strategies for numbers that could mess up options encoding), maybe consider just creating the encoder here instead?

Reply:
yeah, I think that makes sense. The only reason for using the strategies here would be to encode some user data type, and if a user data type ever shows up in an options struct it'll be in the form of a `Document` anyways, meaning they could just encode it themselves. I don't really see us genericizing any options types, which would be the only way a non-`Document` user data element ends up in the options I think.

tl;dr I made it use `BSONEncoder()`